File: java/render/Material.java
public class Material {
  public double ambientWeight;
  public double ambientOcclusionPercent;
  public double diffuseWeight;
  public double specularWeight;
  public double reflectionWeight;  
  public double shininess;
  public double[] diffuseColor;
  public double[] highlightColor;
  public double[] reflectionColor;

  public Material(
      double ambientWeight,
      double ambientOcclusionPercent,
      double diffuseWeight,
      double specularWeight,
      double reflectionWeight,
      double shininess,
      double[] diffuseColor,
      double[] highlightColor,
      double[] reflectionColor) {
    this.ambientWeight = ambientWeight;
    this.ambientOcclusionPercent = ambientOcclusionPercent;
    this.diffuseWeight = diffuseWeight;
    this.specularWeight = specularWeight;
    this.reflectionWeight = reflectionWeight;
    this.shininess = shininess;
    this.diffuseColor = diffuseColor;
    this.highlightColor = highlightColor;
    this.reflectionColor = reflectionColor;    
  }
}

File: js/render/Material.js
Material.prototype.ambientWeight;
Material.prototype.ambientOcclusionPercent;
Material.prototype.diffuseWeight;
Material.prototype.specularWeight;
Material.prototype.reflectionWeight;
Material.prototype.shininess;
Material.prototype.diffuseColor;
Material.prototype.highlightColor;
Material.prototype.reflectionColor;


function Material(
        ambientWeight,
        ambientOcclusionPercent,
        diffuseWeight,
        specularWeight,
        reflectionWeight,
        shininess,
        diffuseColor,
        highlightColor,
        reflectionColor) {
    this.ambientWeight = ambientWeight;
    this.ambientOcclusionPercent = ambientOcclusionPercent;
    this.diffuseWeight = diffuseWeight;
    this.specularWeight = specularWeight;
    this.reflectionWeight = reflectionWeight;
    this.shininess = shininess;
    this.diffuseColor = diffuseColor;
    this.highlightColor = highlightColor;
    this.reflectionColor = reflectionColor;

}
